app/src/test/java/com/x8bit/bitwarden/data/autofill/processor/AutofillProcessorTest.kt

```diff
@@ -82,13 +82,97 @@ class AutofillProcessorTest {
 
     @AfterEach
     fun teardown() {
-        verify(exactly = 1) {
-            dispatcherManager.unconfined
-        }
         unmockkStatic(::createAutofillSavedItemIntentSender)
         unmockkStatic(AutofillRequest.Fillable::toAutofillSaveItem)
     }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `processFillRequest should allow additional requests to be invoked after cancellation signal is triggered`() {
+        // Setup
+        val autofillPartition: AutofillPartition = mockk()
+        val autofillRequest: AutofillRequest.Fillable = mockk {
+            every { packageName } returns PACKAGE_NAME
+            every { partition } returns autofillPartition
+        }
+        val fillRequest: FillRequest = mockk()
+        val saveInfo: SaveInfo = mockk()
+        val filledData = FilledData(
+            filledPartitions = listOf(mockk()),
+            ignoreAutofillIds = emptyList(),
+            originalPartition = mockk(),
+            uri = null,
+            vaultItemInlinePresentationSpec = null,
+            isVaultLocked = false,
+        )
+        val fillResponse: FillResponse = mockk()
+        val cancellationSignalListener = slot<CancellationSignal.OnCancelListener>()
+        every {
+            cancellationSignal.setOnCancelListener(capture(cancellationSignalListener))
+        } just runs
+        every {
+            parser.parse(autofillAppInfo = appInfo, fillRequest = fillRequest)
+        } returns autofillRequest
+        coEvery { filledDataBuilder.build(autofillRequest = autofillRequest) } returns filledData
+        every {
+            saveInfoBuilder.build(
+                autofillAppInfo = appInfo,
+                autofillPartition = autofillPartition,
+                fillRequest = fillRequest,
+                packageName = PACKAGE_NAME,
+            )
+        } returns saveInfo
+        every {
+            fillResponseBuilder.build(
+                autofillAppInfo = appInfo,
+                filledData = filledData,
+                saveInfo = saveInfo,
+            )
+        } returns fillResponse
+        every { fillCallback.onSuccess(fillResponse) } just runs
+
+        // Test
+        autofillProcessor.processFillRequest(
+            autofillAppInfo = appInfo,
+            cancellationSignal = cancellationSignal,
+            fillCallback = fillCallback,
+            request = fillRequest,
+        )
+
+        cancellationSignalListener.captured.onCancel()
+
+        // Test again after cancelling
+        autofillProcessor.processFillRequest(
+            autofillAppInfo = appInfo,
+            cancellationSignal = cancellationSignal,
+            fillCallback = fillCallback,
+            request = fillRequest,
+        )
```

Comment:
Can we add some verifications here so we can see some differences between the first run and the second? I would have excepted to see all the verifications show that 2 things happened instead of 1, but not I'm seeing this is because the dispatcher wasn't told to run, right? It might be more obvious if we are verifying that _nothing_ happened here after we cancel but before we run again.